tools/jdbc/src/main/java/org/duckdb/DuckDBArrowResult.java

```diff
@@ -0,0 +1,66 @@
+package org.duckdb;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+@SuppressWarnings("rawtypes")
+public class DuckDBArrowResult {
+
+	private Class vector_root_class;
+	private Class schema_class;
+	private Class dictionary_class;
+
+	private Class root_allocator_class;
+	private Class buffer_allocator_class;
+
+	private Class c_schema_class;
+	private Class c_array_class;
+	private Class c_data_class;
+	private Class c_dictionary_class;
+
+	private Object allocator;
+
+	private DuckDBResultSet result_set;
+	private Object schema;
+
+	@SuppressWarnings("unchecked")
+	public DuckDBArrowResult(ResultSet result_set_p) throws Exception {
+		if (!(result_set_p instanceof DuckDBResultSet)) {
+			throw new SQLException("This only works with a DuckDB ResultSet");
+		}
+		result_set = (DuckDBResultSet) result_set_p;
+
+		vector_root_class = Class.forName("org.apache.arrow.vector.VectorSchemaRoot");
```

Comment:
For the record, we should really be compiling against the actual classes instead of doing this - it won't cause issues at runtime until this class is loaded if the arrow classes aren't on the classpath

I do understand why you're taking the reflection path fwiw